icechunk/src/change_set.rs

```diff
@@ -24,20 +25,81 @@ pub struct ArrayData {
     pub user_data: Bytes,
 }
 
+impl ManifestSplits {
+    pub fn which_extent(&self, coord: &ChunkIndices) -> SessionResult<&ManifestExtents> {
+        Ok(self.0.get(self.which(coord)?).expect("logic bug"))
+    }
+}
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+pub struct SplitManifest {
+    from: Vec<u32>,
+    to: Vec<u32>,
+    // It's important we keep these sorted, we use this fact in TransactionLog creation
+    chunks: BTreeMap<ChunkIndices, Option<ChunkPayload>>,
+}
+
+impl SplitManifest {
+    pub fn update(&mut self, coord: ChunkIndices, data: Option<ChunkPayload>) {
```

Comment:
I wonder if  this will  be fast enough not to slow down massive virtual dataset ingestion. Maybe keeping a

`range: Vec<(u32,u32)>`

 would be slightly faster?

Reply:
TODO: look for an existing data structure here?